### PR TITLE
Eth deposits

### PR DESCRIPTION
**Pull request by hkalodner**
- Update ArbOS with https://github.com/OffchainLabs/arb-os/pull/260
- Add test for new deposit behavior

### CODE REVIEW

File: docs/ArbOS_Formats.md [hunk 1]
```diff
@@ -107,6 +107,16 @@ Type-specific data:
...
 
+##### Message type 7: Eth deposit transaction
+
+This message type first transfers the given amount of Eth to the sender's account. Then it executes the given transaction.
+
+Type-specific data:
+
+- Unsigned Transaction or Contract transaction as defined in the L2 message section below
+
+This message type is reserved for internal use by ArbOS. It should never appear in the inbox.
```
Comment (edfelten) [inline docs/ArbOS_Formats.md:118]: This one should sometimes appear in the inbox, right?

**Review by edfelten — CHANGES_REQUESTED** [review]
One issue in the format documentation.

File: docs/ArbOS_Formats.md (hunk 1 above)
Comment (hkalodner) [reply docs/ArbOS_Formats.md]: Whoops that was a copy/paste error. I removed that line

**Review by edfelten — APPROVED** [review]
Looks good.

**Pull request merged**